### ISSUE TITLE
False positive no-unused-vars with import used on typescript interfaces

### ISSUE THREAD

**Issue by soullivaneuh** [yannickcr/eslint-plugin-react#2623]
With the following react-native typescript code block:

```typescript
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';

interface Props {
  name: string,
  navigation: StackNavigationProp<any, 'SignUp'>,
}
```

I have the following false positive error:

```
/code/fixtures/App.tsx
   3:10  error  'StackNavigationProp' is defined but never used  no-unused-vars
```

I setup the config for react using the `overrides` section (I have a global and generic configuration, not only for react):

```javascript
overrides: [
  {
    files: ['*.jsx', '*.tsx'],
    extends: [
      'plugin:react/recommended',
    ],
    parser: '@typescript-eslint/parser',
  },
],
```

But the bug happens also with a global configuration.

It looks like the plugin is not working with import used on interfaces.

Note that you may see the full configuration on this Pretty project MR: https://gitlab.com/nexylan/pretty/-/merge_requests/71

**Comment by ljharb** [comment]
Now that `import type` exists, why not use that?

Either way, `no-unused-vars` is an eslint core rule, and not part of this plugin.

**Comment by soullivaneuh** [comment]
@ljharb 

> Now that `import type` exists, why not use that?

Not sure what do you mean by that. May you provide an example?

> Either way, `no-unused-vars` is an eslint core rule, and not part of this plugin.

Well, is was thinking it's related to react specific usage like this rule: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md

Maybe I'm wrong?

Thanks.

**Comment by soullivaneuh** [comment]
This is weird, I tested my config against a `main.ts` file (so not managed by the plugin):

```typescript
import { Bar } from 'bar';

interface Thing {
  foo: Bar;
}
```

And I don't have this king of error.

**Comment by soullivaneuh** [comment]
Ok you was right, this was a typescript plugin issue, adding this rule solve the problem:

```typescript
  '@typescript-eslint/no-unused-vars': ['error', {
    args: 'none',
  }],
```

Thanks for your help. But I'm curious about our first suggestion for `import type`. May you add some details?

**Comment by ljharb** [comment]
Yes, TS 3.8 has `import type` for importing types, which means you can reserve `import` only for value-space imports. It's much clearer and should have been the only way to import types since TS's inception, but at least now they've fixed their mistake.
